Not compatible with pandas==1.4.0

Pandas just released a new version which is not compatible with xtgeo, getting:
```
  >       ier = _cxtgeo.grd3d_inact_outside_pol(
              xc,
              yc,
              self.ncol,
              self.nrow,
              self.nlay,
              self._coordsv,
              self._zcornsv,
              self._actnumsv,  # is modified!
              k1,
              k2,
              iforce,
              method,
          )
  E       TypeError: Cannot cast array data from dtype('O') to dtype('float64') according to the rule 'safe'
```

--- a/src/xtgeo/xyz/_xyz_io.py
+++ b/src/xtgeo/xyz/_xyz_io.py
@@ -330,7 +330,9 @@ def _convert_idbased_xyz(self, df):
 
     idgroups = df.groupby(self._pname)
 
-    newdf = pd.DataFrame(columns=[self._xname, self._yname, self._zname])
+    newdf = pd.DataFrame(
+        columns=[self._xname, self._yname, self._zname], dtype="float64"
+    )
     udef = pd.DataFrame(
         [[999.0, 999.0, 999.0]], columns=[self._xname, self._yname, self._zname]
     )

